staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -770,6 +770,7 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = filterlatency.TrackStarted(handler, "authentication")
 
 	handler = genericfilters.WithCORS(handler, c.CorsAllowedOriginList, nil, nil, nil, "true")
+	handler = genericfilters.WithLongRunningRequestTermination(handler, c.LongRunningFunc, c.terminationSignals.AfterShutdownDelayDuration.Signaled())
```

Comment:
does this mean we close them before non-long-running are finished? Would expect to wait for the latter, then close the former.

Reply:
mhm, I like the idea. I would even extend it to:
- propagate the termination signal (by closing a context) to all long-running requests immediately after receiving the signal
- forcefully kill (after N additional seconds) only if all non-long-running have already finished

Reply:
done